src/components/controls/date-picker.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import styled from 'styled-components';
+import { MdDateRange } from "react-icons/md";
+import { FiSave } from "react-icons/fi";
+
+const FloatDiv = styled.div`
+  float: ${(props) => props.right ? "right" : "left"};
+  max-width: 50%;
+`;
+const TextSpan = styled.span`
+  font-family: ${(props) => props.theme["font-family"]};
+  margin-bottom: 5px;
+  font-size: 12px;
+  font-weight: 400;
+  color: ${(props) => props.theme.color};
+`;
+const IconSpan = styled(TextSpan)`
+  margin: 0 3px;
+  cursor: pointer;
+`;
+
+const DatePicker = ({ value, minDate, maxDate, onChange, right }) => {
+  const [editing, setEditing] = React.useState(false);
+  function processNewValue(newValue) {
+    setEditing(false);
+    onChange(newValue);
+  }
+  function onKeyUp(e) {
+    if (e.key === 'Enter') {
+      processNewValue(e.target.value);
+    }
+  }
+  function onBlur(e) {
+    processNewValue(e.target.value);
+  }
+  let content;
+  if (editing) {
+    const dateInput = (
+      <input
+        type="date"
+        defaultValue={value}
+        min={minDate}
+        max={maxDate}
+        onKeyUp={onKeyUp}
+        onBlur={onBlur}
+      />
+    );
+    const saveIcon = <IconSpan onClick={() => setEditing(false)}><FiSave /></IconSpan>;
+    content = right ? <>{dateInput}{saveIcon}</> : <>{saveIcon}{dateInput}</>;
```

Comment:
The should be tweaked (or perhaps some other element) so that when editing it doesn't appear on the line below the icon when there's enough space for them to be on the same line. I can change this by removing `max-width: 50%` from `<FloatDiv>` but I'm unsure if this is the best without diving deeper.

For instance: 
![image](https://user-images.githubusercontent.com/8350992/137997146-77690af5-6676-4979-80d4-a441784b82fb.png)

Reply:
One of the limitations of the native date picker is that [you can't do much styling](https://stackoverflow.com/questions/14946091/are-there-any-style-options-for-the-html5-date-picker) on the input box as it varies per browser. Seems like, at least for some browsers, the parent div containing icon+input is larger than half the width of the sidebar, and something must be pushed down to a second "row". Here's without `max-width: 50%`:

<img width="239" alt=" 2021-10-19 at 11 52 35 PM" src="https://user-images.githubusercontent.com/13424970/138042625-3919b7a1-a8da-4e6d-b9a5-d722d94b9d79.png">

Reply:
For the record, imposing a `max-width` on the `input` gets ugly:

<img width="231" alt=" 2021-10-19 at 11 58 27 PM" src="https://user-images.githubusercontent.com/13424970/138043413-eeaa628d-f9a0-4b69-a8f3-871fc4a2039e.png">